src/Config/ObjectModel/TelemetryOptions.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config.ObjectModel;
+
+/// <summary>
+/// Represents the options for telemetry.
+/// </summary>
+public record TelemetryOptions(ApplicationInsightsOptions ApplicationInsights)
```

Comment:
How is this level of abstraction going to be utilized when/if we ever support other telemetry mechanisms? Would this record be updated to something like
```csharp
TelemetryOptions(ApplicationInsightsOptions? ApplicationInsights = null, OtherTelemetryProviderOptions providerOptions? = null)

Reply:
Correct, This will be updated based on future requirements. since, currently we don't know what other telemetry Options we would provide support for and how connection to them will be estabilished.